federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -444,20 +451,21 @@ func (secretcontroller *SecretController) reconcileSecret(secret types.Namespace
 }
 
 // delete deletes the given secret or returns error if the deletion was not complete.
-func (secretcontroller *SecretController) delete(secret *apiv1.Secret) error {
-	glog.V(3).Infof("Handling deletion of secret: %v", *secret)
-	_, err := secretcontroller.deletionHelper.HandleObjectInUnderlyingClusters(secret)
+func (secretcontroller *SecretController) delete(obj pkgruntime.Object, namespacedName types.NamespacedName) error {
```

Comment:
Hmm. This is unfortunate that we have to pass two representations of the same object. Perhaps in a future refactor the deletion helper can be adjusted to take the namespaced name instead?